spec/services/decisions/case_decision_tree_spec.rb

```diff
@@ -8,14 +8,24 @@
 
   let(:codefendants_double) { double('codefendants_collection') }
   let(:charges_double) { double('charges_collection') }
+  let(:applicant_double) { double('applicant') }
+  let(:applicant_dob) { Date.new(1990, 1, 1) }
```

Comment:
I suggest to keep the decision trees as clean from any external logic as possible. If we test the service objects separately, then in the decision trees we just need to mock their response but not their implementation.

So for example in the step that uses the `IojPassporter` service object, we can just `allow` it to receive whatever and return true/false. Not worrying about the applicant DOB etc.

Is a suggestion, as we grow in the number of service objects and the decision trees may use more of them, to keep it contained 😄